### PR TITLE
Add Category model and associations

### PR DESCRIPTION
**Pull request by giacoletti**
### This PR contains:
- Creates Category model
- changes add_category_to_articles migration by adding reference to articles table
- adds associations between articles and categories
- fixes all request specs and adds category model specs
- fixes methods in articles_controller.rb

@Mljungho @elvitak @DefyCab 

[Pivotal Tracker Chore](https://www.pivotaltracker.com/story/show/180992131)

### CODE REVIEW

File: app/controllers/api/articles_controller.rb [hunk 1]
```diff
@@ -22,6 +23,8 @@ def show
   def create
     authorize Article.create
     article = Article.create(article_params)
+    article.category = Category.find_by name: params["article"]["category"]
```
Comment (maxarvid) [inline app/controllers/api/articles_controller.rb]: Format your code

File: spec/factories/categories.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :category do
+    name { "MyString" }
```
Comment (maxarvid) [inline spec/factories/categories.rb]: Format your code

**Review by maxarvid — CHANGES_REQUESTED** [review]
See two small comments on single/double quotes in comments.
Not super fond of the changing of the migration, but given that we are preproduction I think we can let it slide.
The PT Chore lacks a link back to the Github PR
Suggestion going forward: You can very easily paste screen grabs in the Conversation part of a PR. For the API this could be the RSpec console output. For the Client you'd want to show off your new feature.
Great work.

File: app/controllers/api/articles_controller.rb (hunk 1 above)
Comment (giacoletti) [reply app/controllers/api/articles_controller.rb:26]: article.category = Category.find_by name: params['article']['category']
this is the end result.
Were the double quotes the issue?

**Review by maxarvid — APPROVED** [review]

**Pull request merged**